classifier.py

```diff
@@ -3402,6 +3403,19 @@ def add(self, classification):
         if not classification.subject.checked: # or self.debug
             classification.subject.assign_to_genre()
 
+        # if classification is genre or NONE from staff, ignore all non-staff genres
+        from model import DataSource
+        staff_data_source = DataSource.lookup(self._db, DataSource.LIBRARY_STAFF)
+        from_staff = classification.data_source == staff_data_source
```

Comment:
Instead of looking up the DataSource object every time you can compare classification.data_source.name against DataSource.LIBRARY_STAFF.